plugins/org.eclipse.fordiac.ide.application/src/org/eclipse/fordiac/ide/application/handlers/RepairCommandHandler.java

```diff
@@ -141,7 +141,9 @@ public boolean performFinish() {
 					handleChangeDataType(choice);
 					break;
 				case CREATE_MISSING_TYPE:
-					ErrorMarkerResolver.repairMissingStructuredDataType(type);
+					if (type instanceof final VarDeclaration varDeclaration) {
+						ErrorMarkerResolver.repairMissingStructuredDataType(varDeclaration);
+					}
```

Comment:
should there be some info to the user in the else?

Reply:
i think we should not even be able to open the dialog at this point if the user selects an error marker which is not a "Missing DT/FBT".  
This is verified in the repairProperty tester.
But in general the repairCommandHandler needs a clean-up.